packages/go_router/example/lib/main.dart

```diff
@@ -5,89 +5,93 @@
 import 'package:flutter/material.dart';
 import 'package:go_router/go_router.dart';
 
-// This scenario demonstrates a simple two-page app.
-//
-// The first route '/' is mapped to Page1Screen, and the second route '/page2'
-// is mapped to Page2Screen. To navigate between pages, press the buttons on the
-// pages.
-//
-// The onPress callbacks use context.go() to navigate to another page. This is
-// equivalent to entering url to the browser url bar directly.
+/// This sample app shows an app with two screens.
+///
+/// The first route '/' is mapped to [HomeScreen], and the second route
+/// '/details' is mapped to [DetailsScreen].
+///
+/// The buttons use context.go() to navigate to each destination. On mobile
+/// devices, each destination is deep-linkable and on the web, can be navigated
+/// to using the address bar.
+void main() => runApp(const MyApp());
 
-void main() => runApp(App());
+/// The route configuration.
+final GoRouter _router = GoRouter(
```

Comment:
should we advocate creating GoRouter as a global singleton? This makes sense to me,  and I can't think of anything would break b/c this.

Reply:
I filed an issue so we can discuss: https://github.com/flutter/flutter/issues/115379